tfs/src/main/java/hudson/plugins/tfs/TeamBuildEndpoint.java

```diff
@@ -196,22 +197,25 @@ void dispatch(final StaplerRequest req, final StaplerResponse rsp, final TimeDur
      * multibranch pipeline projects in this case.
      *
      * We will try to determine the branch name in the following sequence:
-     * 1. Check if the jobName is composed from ${multibranch_pipeline}/${branch_name}
-     * 2. Check if the payload has BuildSource variable defined (for PR builds)
+     * 1. For JenkinsQueueJob task 1.115.0+, we send the branch in "QueueJobTask.MultibranchPipelineBranch".
+     * 2. Check if the jobName is composed from ${multibranch_pipeline}/${branch_name}
+     * 3. Check if the payload has BuildSource variable defined (for PR builds)
      *
      * If we can't determine the branch name, throw.
      */
     private String getBranch(final String jobName, final StaplerRequest req) {
-        String sourceBranch = null;
+        final String json = req.getParameter("json");
+        final JSONObject formData = JSONObject.fromObject(json);
+        final TeamBuildPayload payload = EndpointHelper.MAPPER.convertValue(formData, TeamBuildPayload.class);
 
-        if (jobName.indexOf('/') > 0) {
-            sourceBranch = jobName.substring(jobName.indexOf('/') + 1);
-        } else {
-            final String json = req.getParameter("json");
-            final JSONObject formData = JSONObject.fromObject(json);
-            final TeamBuildPayload payload = EndpointHelper.MAPPER.convertValue(formData, TeamBuildPayload.class);
+        String sourceBranch = payload.BuildVariables.get(QUEUEJOBTASK_MULTIBRANCH_JOB_BRANCH);
 
-            sourceBranch = payload.BuildVariables.get(BUILD_SOURCE_BRANCH);
+        if (sourceBranch == null || sourceBranch.trim().isEmpty()) {
+            if (jobName.indexOf('/') > 0) {
+                sourceBranch = jobName.substring(jobName.indexOf('/') + 1);
+            } else {
+                sourceBranch = payload.BuildVariables.get(BUILD_SOURCE_BRANCH);
+            }

```

Comment:
Not important, but consider saving the result of `jobName.indexOf('/')` instead of calling it twice.